### ISSUE TITLE
Parent Issue: Display ads on Front Pages

### ISSUE THREAD

**Issue by tochwill** [bbc/simorgh#6475]
### **Documentation**
[Product/UX/Dev Documentation Notes and Links](https://paper.dropbox.com/doc/Display-Ads-on-Front-Page-ProductUXDev-Documentation-Notes--A0NGJDTHGFcDah2rTZC8HmqbAg-KaLxGOm5QR5kyqzEGYAiF)

**Requirements**
We need to show adverts on front pages for some services. They need to work on both Canonical and AMP versions of the page. Further details on these requirements are explained next to the relevant issues below, and can be found on confluence, at `/WSFrontPGS/Adverts`.

### Decide whether an Ad Slot should be rendered, and render if so

![Ads Toggle Flow](https://user-images.githubusercontent.com/6690955/87280068-a1406f00-c4e9-11ea-92ce-9f9ddbbbded7.png)

### Release 1: Canonical
**Determine whether and ad should be rendered**
- [x] Check iSite Toggle https://github.com/bbc/simorgh/issues/6866
- [x] Ensure Config Service is production ready
- [x] Ensure Config Service is understood by Simorgh developers @tochwill arrange a short session by 15th August - and ensure relevant people have access by 11th August.

https://paper.dropbox.com/doc/Enabling-Remote-Toggles-in-Simorgh--A4m2kw5A_v9Ap9EmPzI7vLW5Ag-vCtOnfTN7vobbeBwsj3b2
- [x] Check User Location https://github.com/bbc/simorgh/issues/7066
  - [x] Pass headers from Mozart to Simorgh: https://github.com/bbc/mozart-routing/pull/1473/
  - [x] Read these header values in Simorgh: https://github.com/bbc/simorgh/pull/7205
  - [x] Vary on these headers in Mozart: https://github.com/bbc/mozart-routing/pull/1515

**Ensure correct browser support**
- [x] Load Ads script with support for older browsers: https://github.com/bbc/simorgh/issues/7313
- [x] Ensure Promise.finally issue with GNL script is sorted: https://github.com/bbc/simorgh/issues/7282

**Reserve the correct amount of space for each ad slot**
- [x] Ensure Ads match UX specifications: https://github.com/bbc/simorgh/issues/7119, https://github.com/bbc/simorgh/pull/7537

**Cleanup work**
- [x] Remove 'hasAds' from config: https://github.com/bbc/simorgh/pull/7299
- [x] Remove debug cookie: https://github.com/bbc/simorgh/pull/7446

**A11y and UX review**
- [x] A11y review + UX review (#5956) (A11y requirements are the same for Leaderboard and MPU) - ensure this issue contains all of the a11y acceptance criteria.

**Monitoring and Error Reporting**
- [x] Update runbook to explain toggles, including a section on what to check if ads aren't displaying: https://github.com/bbc/simorgh-infrastructure/issues/1124. Include references to CSP violation monitoring that is already in place.
- [ ] Create issue to add client error monitoring to Simorgh. This will not block our release, and will be handed over to the teams remaining on world service.

**Content Security Policy**
- [x] Ensure LIVE CSP is updated correctly and tested before enabling ads on live environment https://github.com/bbc/simorgh/issues/7366

**Rollout**
- [x] We should rollout display ads service by service using iSite: https://github.com/bbc/simorgh-infrastructure/issues/1081

**Testing Notes**
- [x] Ensure integration tests are written: https://github.com/bbc/simorgh/issues/6255 @PriyaKR will arrange something by 15th August.

- [x] Take acceptance criteria from Miro and formulate a test plan that takes into account the testing pyramid.
- [ ] Load pages with ads on slow connections to observe any page content-shifting for AMP/Canonical pages. Discuss mitigation of such content-shifting:
        - Best practice with respect to loading ads (AMP/Canonical - can we use WebVitals to monitor and optimise this? Largest Contentful Paint (LCP), First Input Delay (FID), Cumulative Layout Shift (CLS))
- [x] UX + A11y review.
- [x] Ensure page works with ad blockers on.
- [x] GNL testing.
- [x] We should test each service is working as expected.
- [x] We should perform extensive cross browser testing for this release.

### Release 2 - Comscore on canonical (this may be finished before Release 1 above).
- [x] Implement Comscore tracking for Canonical (#6754).
- [x] Toggle Comscore tracking on LIVE (https://github.com/bbc/simorgh/issues/7483)
- [ ] Remain in touch with Comscore to finalise how we are notified about updates to the script. @DenisHdz has been in contact and can follow this up.

### Third Release - AMP
**Determine whether leaderboard Ad should be rendered - AMP**
- [x] Use amp-geo to locate user, and only show ad outside of UK. https://github.com/bbc/simorgh/pull/6961
- [x] Handle content shifting https://github.com/bbc/simorgh/issues/7380
- [ ] Define (and implement) monitoring and error reporting required for AMP ads.

**AMP Consent**
- [ ] User should be able to revoke consent on AMP pages. https://github.com/bbc/simorgh/issues/7086

**Comscore AMP**
- [x] Implement Comscore tracking for AMP (#6472).
- [x] Update CSP for Comscore on AMP and enable on live (#7202)

****
**Issues sitting with the GNL team**
- [x] GNL to remove need for query string in Canonical implementation.
- [x] GNL UX change for grey background colour consistency (or AMP change to match Canonical).
        - UX are happy for us to change our AMP to match Canonical https://github.com/bbc/simorgh/issues/7119
- [x] GNL change to apply hyperlink to ‘Advertisement’ label on Canonical.
- [x] GNL - Advertisement link goes to `http` rather than `https` http://www.bbc.com/usingthebbc/cookies/
- [x] E2E tests fail on Zhongwen (investigation #6734 to determine which code needs updating)
- [x] GNL - Query parameter `?site=test` needs to be enabled.

**Post Release**
- [ ] Once OTG work to update location headers is complete (Link to JIRA in Ads Overview Paper Doc), we can then pass the geolocation data to GNL for canonical ads from Simorgh, and they can stop their making a call from the client.
- [ ] Consider using https://www.npmjs.com/package/@bbc/psammead-content-anchor to adjust the height and width of the ad slot offscreen once the ad has loaded in
- [ ] What happens if we navigate between two pages with ads on the client, or between two services on the client, with or without ads? Ensure there is an issue to cover these scenarios, and hand this over to be worked on after the initial release.
- [ ] Client error monitoring should be handed over.

**Comment by mmcculloch** [comment]
After resolving some issues I've raised a PR to enable Canonical ads again on test. https://github.com/bbc/simorgh/pull/6690

Some side effects:

- GNL Ad debugging is enabled for Canonical in order to disable geoIp check. This may produce debug information on test, but will not go live.
- I had to disable ads on `zhongwen` since the e2e tests were failing for an unknown reason. This could be a bug in GNL as I can't think of any reason for it. I've raised an issue to investigate this https://github.com/bbc/simorgh/issues/6734
